engine/class_modules/priest/sc_priest_pets.cpp

```diff
@@ -1155,6 +1264,43 @@ void priest_t::trigger_shadowflame_prism( player_t* target )
   }
 }
 
+void priest_t::trigger_living_shadow_action( player_t* target, living_shadow_action action )
+{
+  priest_pet_t* pet = debug_cast<priest_pet_t*>( pets.your_shadow.active_pet() );
+
+  if ( pet )
+  {
+    switch ( action )
+    {
+      case living_shadow_action::SHADOW_SPIKE:
+        assert( pet->shadow_spike );
+        pet->shadow_spike->set_target( target );
```

Comment:
Create a function to select the correct action_t* and then have the set_target and execute lines a single time.

Reply:
Good shout, added this.
Neatens it up and makes it easier in the future if they add anything.